routes/shopItem.js

```diff
@@ -0,0 +1,24 @@
+const express = require("express");
+const fs = require("fs");
+const resolve = require("path").resolve;
+let id = 0;
+
+shopItemRouter = express.Router();
+
+shopItemRouter.use("/", (req,res,next) => {
+id++;
+req.body.id = id;
+next();
+})
```

Comment:
This isn't needed for PUT. In that case the id field should already present in the body (since the server need to know the item ID to update/overwrite). I suggest to just increase the ID in the post handler, and add the ID there as well.